src/config/config_type.rs

```diff
@@ -48,19 +48,21 @@ impl ConfigType {
     /// It returns both the deserialized, as well as, a JSON representation
     /// of the provided config.
     /// It returns an error if any of the serialization or deserialization steps fail.
-    pub fn deserialize<Static, Dynamic>(
+    pub fn deserialize<TextConfiguration, BinaryConfiguration>(
         self,
         filter_name: &str,
-    ) -> Result<(serde_json::Value, Static), Error>
+    ) -> Result<(serde_json::Value, TextConfiguration), Error>
     where
-        Dynamic: prost::Message + Default,
-        Static: serde::Serialize
-            + for<'de> serde::Deserialize<'de>
-            + TryFrom<Dynamic, Error = ConvertProtoConfigError>,
+        BinaryConfiguration: prost::Message + Default,
+        TextConfiguration:
```

Comment:
`StringConfiguration` ? (Rather than `TextConfiguration`)?

Just a possible suggestion, non-blocking.

Reply:
I chose `Text` over `String` because `String` is a specific type in Rust as opposed to just "human readable", and it's shorter.